### DISCUSSION TITLE
java.lang.NoClassDefFoundError: io/netty/internal/tcnative/SSLSessionCache

### DISCUSSION THREAD

**Discussion by Akaame** [LogNet/grpc-spring-boot-starter#278] [Q&A] [answered]
Hello,

Starter project has the following dependency which seems to be incompatible with `grpc-1.43.2`: `compile 'io.netty:netty-tcnative-boringssl-static:2.0.25.Final'`

During initialization we get `java.lang.NoClassDefFoundError: io/netty/internal/tcnative/SSLSessionCache` while Spring performs autoconfiguration. This error is then resolved by excluding the tcnative and explicitly adding the dependency with a newer version (currently `2.0.47.Final`).

SSLSessionCache itself seems to be added in `2.0.31.Final` from what I have seen.

Is this the recommended way to use the starter project (given that you also exclude tcnative in the demo project)?
Why is not tcnative upgraded to a more recent version?

Thanks in advance and cheers.

**Comment by jvmlet** [comment, marked as answer]
The latest released version `4.5.10` uses grpc `1.42.0`. I'll check the tcnative version that should be used with `1.43.2`.
`4.6.10` will be released soon with `1.43.2`

**Reply by Akaame** [reply]
Sorry, this was an overlook from my side. Another library was using `1.43.2` so it was preferred over `1.42.0`.

Thanks and cheers.

**Comment by jvmlet** [comment]
You are welcome to upgrade to [ 4.6.0](https://github.com/LogNet/grpc-spring-boot-starter/blob/master/ReleaseNotes.md#version-460)
